src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -704,84 +702,80 @@ private static bool IsNodeOrAnyAncestorLeftSideOfDot(SyntaxNode node, ISyntaxFac
             {
-                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingType);
+                additionalProperties.Add(containingTypeProperty);
             }
 
-            if (GetInfo(syntaxFacts.GetContainingMemberDeclaration(node, node.SpanStart),
-                ContainingMemberInfoPropertyName,
-                semanticModel,
-                out var findUsagePropertyForContainingMember))
+            if (TryGetAdditionalProperty(
+                    syntaxFacts.GetContainingMemberDeclaration(node, node.SpanStart),
+                    ContainingMemberInfoPropertyName,
+                    semanticModel,
+                    out var containingMemberProperty))
             {
-                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingMember);
+                additionalProperties.Add(containingMemberProperty);
             }
 
-            return additionalProperties.ToImmutableAndFree();
+            return additionalProperties.ToImmutable();
         }
 
-        internal static ImmutableArray<FindUsageProperty> GetAdditionalFindUsagesProperties(ISymbol definition)
+        internal static ImmutableDictionary<string, string> GetAdditionalFindUsagesProperties(ISymbol definition)
         {
-            var additionalProperties = new ArrayBuilder<FindUsageProperty>();
+            var additionalProperties = ImmutableDictionary.CreateBuilder<string, string>();
 
             var containingType = definition.ContainingType;
             if (containingType != null &&
-                TryGetAdditionalProperty(ContainingTypeInfoPropertyName, containingType, out var findUsagePropertyForContainingType))
+                TryGetAdditionalProperty(ContainingTypeInfoPropertyName, containingType, out var containingTypeProperty))
             {
-                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingType);
+                additionalProperties.Add(containingTypeProperty);
             }
 
             var containingSymbol = definition.ContainingSymbol;
 
             // Containing member should only include fields, properties, methods, or events.  Since ContainingSymbol can return other types, use the return value of GetMemberType to restrict to members only.)
             if (containingSymbol != null &&
                 containingSymbol.GetMemberType() != null &&
-                TryGetAdditionalProperty(ContainingMemberInfoPropertyName, containingSymbol, out var findUsagePropertyForContainingMember))
+                TryGetAdditionalProperty(ContainingMemberInfoPropertyName, containingSymbol, out var containingMemberProperty))
             {
-                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingMember);
+                additionalProperties.Add(containingMemberProperty);
             }
 
-            return additionalProperties.ToImmutableAndFree();
+            return additionalProperties.ToImmutable();
```

Comment:
Why the change from `.ToImmutableAndFree` to `.ToImmutable`?

Reply:
I switched from using a pooled ArrayBuilder to ImmutableDictionary.Builder as we now return a dictionary instead of an array of key-value pairs.